src/Modules/ModulesGraded.jl

```diff
@@ -0,0 +1,446 @@
+    @assert length(d) == rank(F)
+    r = new{elem_type(base_ring(F))}(F, d)
+    return r
+  end
+
+  function FreeMod_dec{T}(R::CRing_dec,S::Vector{Symbol},d::Vector{GrpAbFinGenElem}) where T <: CRingElem_dec
+    r = new{elem_type(R)}()
+    r.F = FreeMod{T}(length(d),R,S)
+    r.d = d
+    return r
+  end
+end
+
+@doc Markdown.doc"""
+    FreeMod_dec(R::CRing_dec, n::Int, name::String = "e"; cached::Bool = false) 
+
+Construct a decorated (graded or filtered) free module over the ring `R` with rank `n`
+with the standard degrees, that is the standard unit vectors have degree 0.
+Additionally one can provide names for the generators. If one does 
+not provide names for the generators, the standard names e_i are used for 
+the standard unit vectors.
+"""
+function FreeMod_dec(R::CRing_dec, n::Int, name::String = "e"; cached::Bool = false) 
+  return FreeMod_dec{elem_type(R)}(R, [Symbol("$name[$i]") for i=1:n], [decoration(R)[0] for i=1:n])
+end
+
+@doc Markdown.doc"""
+    free_module_dec(R::CRing_dec, n::Int, name::String = "e"; cached::Bool = false)
+
+Create the decorated free module $R^n$ equipped with its basis of standard unit vectors
+and standard degrees, that is the standard unit vectors have degree 0.
+
+The string `name` specifies how the basis vectors are printed. 
+"""
+free_module_dec(R::CRing_dec, n::Int, name::String = "e"; cached::Bool = false) = FreeMod_dec(R, n, name, cached = cached)
+
+
+@doc Markdown.doc"""
+    FreeMod_dec(R::CRing_dec, d::Vector{GrpAbFinGenElem}, name::String = "e"; cached::Bool = false) 
+
+Construct a decorated (graded or filtered) free module over the ring `R` 
+with rank `n` where `n` is the length of `d`. `d` is the vector of degrees for the 
+components, i.e. `d[i]` is the degree of `e[i]` where `e[i]` is the `i`th standard unit
+vector of the free module.
+Additionally one can provide names for the generators. If one does 
+not provide names for the generators, the standard names e_i are used for 
+the standard unit vectors.
+"""
+function FreeMod_dec(R::CRing_dec, d::Vector{GrpAbFinGenElem}, name::String = "e"; cached::Bool = false) 
+  return FreeMod_dec{elem_type(R)}(R, [Symbol("$name[$i]") for i=1:length(d)],d)
+end
+
+@doc Markdown.doc"""
+    free_module_dec(R::CRing_dec, d::Vector{GrpAbFinGenElem}, name::String = "e"; cached::Bool = false)
+
+Create the decorated free module $R^n$ (`n` is the length of `d`)
+equipped with its basis of standard unit vectors where the 
+i-th standard unit vector has degree `d[i]`.
+
+The string `name` specifies how the basis vectors are printed. 
+"""
+free_module_dec(R::CRing_dec, d::Vector{GrpAbFinGenElem}, name::String = "e"; cached::Bool = false) = FreeMod_dec(R, d, name, cached = cached)
+
+
+function FreeMod_dec(F::FreeMod, d::Vector{GrpAbFinGenElem})
+  return FreeMod_dec{elem_type(base_ring(F))}(F, d)
+end
+
+
+function AbstractAlgebra.extra_name(F::FreeMod_dec)
+  t = Hecke.get_special(F, :twist)
```

Comment:
This also must be changed to use `get_attribute` instead. The CI tests only pass because you don't actually test this code ...